### PR TITLE
Feature 897 sct support for unity on csm operator

### PR DESCRIPTION
**Pull request by rajendraindukuri**
# Description
Added SCT support for Unity XT on csm-operator

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/897|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Tested SCT feature for unity on csm-operator by installing the csm-operator build with SCT changes and creating a Unity CR using the Unity build with SCT changes. Test results look good

image

image

### CODE REVIEW

File: operatorconfig/driverconfig/unity/v2.8.0/controller.yaml [hunk 1]
```diff
@@ -71,6 +71,16 @@ rules:
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
     verbs: ["update", "patch"]
+  # Permissions for CSIStorageCapacity
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["csistoragecapacities"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  - apiGroups: [""]
+    resources: ["pods"]
```
Comment (rensyct) [inline operatorconfig/driverconfig/unity/v2.8.0/controller.yaml:79]: Permissions for pods are already there in Line No 43. Please check

File: tests/config/driverconfig/unity/v2.8.0/controller.yaml [hunk 2]
```diff
@@ -71,6 +71,16 @@ rules:
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
     verbs: ["update", "patch"]
+  # Permissions for CSIStorageCapacity
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["csistoragecapacities"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  - apiGroups: [""]
+    resources: ["pods"]
```
Comment (rensyct) [inline tests/config/driverconfig/unity/v2.8.0/controller.yaml:79]: same comment as above

File: samples/storage_csm_unity_v280.yaml [hunk 3]
```diff
@@ -79,6 +84,11 @@ spec:
       - name: external-health-monitor
         enabled: false
         args: ["--monitor-interval=60s"]
+      # Uncomment the following to configure how often external-provisioner polls the driver to detect changed capacity
```
Comment (rensyct) [inline samples/storage_csm_unity_v280.yaml:87]: Please add this block in all the samples that are updated to maintain consistency..  the one under config and the one under e2e

**Review by rensyct — COMMENTED** [review]
Please sign all your commits

File: operatorconfig/driverconfig/unity/v2.8.0/controller.yaml (hunk 1 above)
Comment (rajendraindukuri) [reply operatorconfig/driverconfig/unity/v2.8.0/controller.yaml]: done

File: samples/storage_csm_unity_v280.yaml (hunk 3 above)
Comment (rajendraindukuri) [reply samples/storage_csm_unity_v280.yaml:87]: done

**Review by rensyct — APPROVED** [review]

**Review by panigs7 — APPROVED** [review]

**Pull request merged**